src/arch/riscv/interrupts.c

```diff
@@ -68,15 +78,34 @@ void interrupts_arch_enable(irqid_t int_id, bool en)
 
 void interrupts_arch_handle(void)
 {
+#if (IRQC == AIA)
+    unsigned long stopi = csrs_stopi_read();
+
+    stopi = stopi >> TOPI_IID_SHIFT;
+    switch (stopi) {
+        case IRQ_S_SOFT:
+            interrupts_handle(interrupts_ipi_id);
+            csrs_sip_clear(SIP_SSIP);
+            break;
+        case IRQ_S_TIMER:
+            interrupts_handle(irqc_timer_int_id);
+            break;
+        case IRQ_S_EXT:
+            irqc_handle();
+            break;
+        default:
+            break;
```

Comment:
Shouldn't this print warning?

Reply:
Agree!
Please take a look [here](https://github.com/D3boker1/bao-hypervisor/blob/bd10f0347eefad0e3f21652d51d09c8eb40979be/src/arch/riscv/interrupts.c#L97)